aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ImmutabilityHelper.java

```diff
@@ -29,6 +32,14 @@ static boolean isUpgradeToOracleSE2(final ResourceModel previous, final Resource
                 ORACLE_SE2.equalsIgnoreCase(desired.getEngine());
     }
 
+    static boolean isOracleConvertToCDB(final ResourceModel previous, final ResourceModel desired) {
+        return previous.getEngine() != null &&
+                (
+                        (ORACLE_EE.equalsIgnoreCase(previous.getEngine()) && ORACLE_EE_CDB.equalsIgnoreCase(desired.getEngine()) ||
```

Comment:
engine sometimes can be omitted in customer's template. E.g. when customer is restoring the instance from DB or cluster snapshot, performing Point in Time restore or creating a read replica. Were such scenarios tested?